biz.aQute.bndlib/src/aQute/bnd/osgi/URLResource.java

```diff
@@ -55,7 +55,7 @@ private ByteBuffer getBuffer() throws Exception {
 	private InputStream open() throws Exception {
 		URLConnection conn;
 		InputStream in;
-		if (client != null) {
+		if (client != null && !"jrt".equals(url.getProtocol())) {
```

Comment:
I would instead suggest moving the test to the constructor:

```java
this.client = "jrt".equals(url.getProtocol()) ? null : client;
```

Then the test is not done each open call and the object does not hold a HttpClient reference it will never use.

Reply:
yes of course